### DISCUSSION TITLE
GNU Make - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#146] [Announcements]
# GNU Make - CS自学指南

CS自学指南

https://csdiy.wiki/%E5%BF%85%E5%AD%A6%E5%B7%A5%E5%85%B7/GNU_Make/

**Comment by oscarshan** [comment]
你好，CS61A 这门课现在学不了了吗？

**Reply by UMARU486** [reply]
+1  在2022 spring的网站上点击recording之后进入油管，但显示是“私享视频”，无法播放…

**Reply by UMARU486** [reply]
解决了  在知乎的一篇帖子里找到2020 summer的链接了

知乎搜 cs61A,第一篇帖子就有

**Reply by oscarshan** [reply]
感谢呢，找到了!

**Reply by alinechiang** [reply]
https://inst.eecs.berkeley.edu/~cs61a/fa20/

**Reply by MotokoISL** [reply]
感恩。

**Comment by viewray** [comment]
https://www.youtube.com/c/JohnDeNero/videos

**Comment by MAISwords** [comment]
作为一名26岁刚转计算机这个行业的新人，感谢你的分享。

**Comment by GitHub4zhuyu** [comment]
希望有朝一日，我也可以向你一样，帮助这么多人。

**Comment by khoryz666** [comment]
谢谢分享 您的大恩我感激不尽！自学者的福音！

**Comment by Healer-heal** [comment]
感谢前辈！！！

**Comment by hg-gong** [comment]
前言写到心坎里去了，作者可以发一个eth地址嘛，这么好的资源，强烈要求打赏~~~

**Reply by PKUFlyingPig** [reply]
哈哈不必了，知识无价～～

**Reply by 325146** [reply]
作为二本大一学生在课堂上看到这句话多少有点热泪盈眶🥹

**Comment by cxchhh** [comment]
#感谢前辈分享！

**Reply by elain-tang** [reply]
捉学长（

**Comment by ghost** [comment]
请问：自学的坏处一节中，DDL 是指什么呢？谢谢~~

**Reply by cxchhh** [reply]
deadline，最后截止日期

**Reply by ghost** [reply]
原来是这样啊！非常感谢！

**Comment by darkbluesoul** [comment]
非常感谢，引路人！

**Comment by haoy-zzz** [comment]
请问交流群的方法具体怎么找到，不太明白是什么意思

**Reply by calyle** [reply]
我觉得您应该好好看看作者所提到的[提问的智慧](https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way/blob/main/README-zh_CN.md):smile:

**Comment by comi-zhang** [comment]
谢谢博主，虽然我啃得很艰难，但我认为你的教程很好地帮我改正了知识化碎片学习的缺点

**Comment by beautyseeker** [comment]
MakeFile的本质是在描述文件依赖关系，以及依赖文件变动时的自动执行程序规则。
例如目标文件target的生成依赖于一堆dependancies，每次depandancies只要发生改动就会自动执行commands
target : dependancies
    commands
Makefile 有诸多简写语法、隐式推导规则和内置变量用法，有印象即可需要时再查。

**Comment by qiaozhiyi** [comment]
我们老师没讲过gcc
2233

**Comment by Promin3** [comment]
makefile 的教程我觉得文档还是写得有点多，于是我翻译了一个更简短的教程和 cheatsheet。https://github.com/Promin3/Makefile-tutorial-Chinese

**Reply by CodeCanvas-Ak** [reply]
万分感谢！

**Comment by hikomoon** [comment]
我最近 rust 写的比较多，如果打算用 rust 作为主力系统编程语言的同学，可以看看 justfile :)
